### DISCUSSION TITLE
Model response disappears as it's being written and is replaced by error message. On refresh, the whole message can be found.

### DISCUSSION THREAD

**Discussion by erickrf** [danny-avila/LibreChat#4346] [Troubleshooting]
### What happened?

Sometimes, when the model is still writing a response, it disappears completely and is replaced by an error message saying

```
Error connecting to server, try refreshing the page.
```

Refreshing the page actually works, I can later see the message. In the logs, I can see 

```
[OpenAIClient.chatCompletion][finalChatCompletion] Aborted Message
[OpenAIClient.chatCompletion][finalChatCompletion] API error
```

Considering that I saw it happen with very long messages, it may have happened that the endpoint couldn't finish the message within the given maximum number of tokens and something weird happened.

After that, it is still possible to continue the conversation, but each new message produced by the model erases all the previous ones from the chat. Again, refreshing the page fixes it and shows them.

This happened to me using a LiteLLM endpoint.

### Steps to Reproduce

1. Write any prompt to a model
2. Keep click on "Edit" and "Save & submit" the model answer, to make it longer and longer.

### What browsers are you seeing the problem on?

Firefox, Chrome

### Relevant log output

```shell
[OpenAIClient.chatCompletion][finalChatCompletion] Aborted Message
[OpenAIClient.chatCompletion][finalChatCompletion] API error
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SailFlorve** [comment]
Sometimes I encounter this issue too! While chatting, suddenly only the last message remains on the page. Refreshing it can restore the conversation.

**Comment by diemus** [comment]
same, I have checked the code , not found a clue why

**Comment by danny-avila** [comment]
> Considering that I saw it happen with very long messages

the issue stems from client <> server connection. If you are hosting the instance behind a reverse proxy like NGINX, it has a default connection timeout of 1 minute. This is what's likely happening, especially since you see `Error connecting to server, try refreshing the page.`, which is usually shown for a connection issue.

**Reply by erickrf** [reply]
That's correct, further investigation showed that it had to do with a connection timeout handed by a proxy. 

Still, I believe that LibreChat's behavior here is buggy, I'd expect the truncated message to be still visible, as well as the conversation to continue normally after a refresh.
